bff/src/Altinn.Authentication.UI/Altinn.Authentication.UI/Controllers/SystemUserController.cs

```diff
@@ -129,10 +130,15 @@ public async Task<ActionResult> Post([FromBody] SystemUserDescriptor newSystemUs
         // Get the partyId from the context (Altinn Part Coook)
         int partyId = AuthenticationHelper.GetRepresentingPartyId( _httpContextAccessor.HttpContext!);
         newSystemUserDescriptor.OwnedByPartyId = partyId.ToString();
-        SystemUser? systemUser = await _systemUserService.CreateSystemUser(partyId, newSystemUserDescriptor, cancellationToken);
-        if (systemUser is not null)
+        Result<SystemUser> systemUser = await _systemUserService.CreateSystemUser(partyId, newSystemUserDescriptor, cancellationToken);
+        if (systemUser.IsSuccess)
         {           
-            return Ok(systemUser);
+            return Ok(systemUser.Value);
+        }
+
+        if (systemUser.IsProblem)
```

Comment:
You don't need this `if` - `Result` is either success or problem. It should never be neither.

Reply:
fixed